### PR TITLE
Added support of active projects exception

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Review by aniezurawski — CHANGES_REQUESTED** [review]

File: src/neptune/management/exceptions.py [hunk 1]
```diff
@@ -201,3 +201,8 @@ def __init__(self, **kwargs):
         else:
             modified_kwargs["requested"] = '"' + requested + '"'
         super().__init__(**modified_kwargs)
+
+
+class ActiveProjectsLimitReachedException(ManagementOperationFailure):
+    code = 26
+    description = "Limit of active projects reached. You can have up to {currentQuota} active projects simultaneously."
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply src/neptune/management/exceptions.py:208]: Not in management API :wink:

File: src/neptune/internal/backends/swagger_client_wrapper.py [hunk 2]
```diff
@@ -82,6 +83,9 @@ def handle_neptune_http_errors(response, exception: Optional[HTTPError] = None):
             "WORKSPACE_IN_READ_ONLY_MODE": lambda response_body: NeptuneLimitExceedException(
                 reason=response_body.get("title", "Unknown reason")
             ),
+            "LIMIT_OF_ACTIVE_PROJECTS_REACHED": lambda response_body: ActiveProjectsLimitReachedException(
+                currentQuota=response_body.get("currentQuota", "<unknown key>")
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply src/neptune/internal/backends/swagger_client_wrapper.py]: Done

File: src/neptune/management/exceptions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Raalsky) [reply src/neptune/management/exceptions.py:208]: I remember that the main intention was to prepare some "easily processable" exceptions as they'll be mostly called in CI/CD workflows not by the users themselves. I'm not sure how many of these assumptions are up-to-date.

[the remark this answers is not in the document]
Comment (aniezurawski) [reply src/neptune/management/exceptions.py:208]: Good error message and some `code` to be checked programmatically are not mutually exclusive. ;)

[the remark this answers is not in the document]
Comment (normandy7) [reply src/neptune/management/exceptions.py:208]: If we want to be more helpful, we can add more context and "what to do next", sure.

Is this exception thrown when you try `create_project()` but your quota is full?

[the remark this answers is not in the document]
Comment (Raalsky) [reply src/neptune/management/exceptions.py:208]: > Is this exception thrown when you try create_project() but your quota is full?

Yes

[the remark this answers is not in the document]
Comment (normandy7) [reply src/neptune/management/exceptions.py:208]:
```suggestion
    description = (
        "Limit of active projects reached. You can have up to {currentQuota} active projects simultaneously. "
        "To create a new project, you need to either archive an active project or increase the quota of active "
        "projects in the workspace."
    )
```

**Review by normandy7 — COMMENTED** [review]
Proposed slightly more elaborate message.

**Review by normandy7 — APPROVED** [review]

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**